### ISSUE TITLE
Version 2.7.1 is breaking class names in SCSS

### ISSUE THREAD

**Issue by stevenkkim** [electron-userland/electron-webpack#296]
I updated from 2.6.2 to 2.7.1 and my styles broke during compilation.
I'm using SCSS.
Version 2.6.2 works fine.

I have 3 SCSS files:
```
styles/bootstrap.scss
styles/styles.scss
styles/react-contextmenu.scss
```

which I import in my react component componets/App.js as follows:
```
import '../styles/bootstrap.scss';
import '../styles/styles.scss';
import '../styles/react-contextmenu.scss';
```

I have installed `node-sass` and `sass-loader` per these instructions: https://webpack.electron.build/add-ons#sassscss

FYI, I am customizing my bootstrap.scss, which may be causing problems. bootstrap.scss looks like this:

```
// https://getbootstrap.com/docs/4.3/getting-started/theming/
$font-size-base: 0.875rem;
@import "../../node_modules/bootstrap/scss/bootstrap";

```

With 2.7.1, a styles.css file is generated, but all the class names have been replaced with gibberish.

For example, in 2.6.2 the compiled styles.css for heading looks like this:

```
h1, h2, h3, h4, h5, h6,
.h1, .h2, .h3, .h4, .h5, .h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2; }

h1, .h1 {
  font-size: 2.1875rem; }

h2, .h2 {
  font-size: 1.75rem; }

h3, .h3 {
  font-size: 1.53125rem; }

h4, .h4 {
  font-size: 1.3125rem; }

h5, .h5 {
  font-size: 1.09375rem; }

h6, .h6 {
  font-size: 0.875rem; }
```

But in 2.7.1, it looks like this:

```
h1, h2, h3, h4, h5, h6,
._6P1pAC-zOqNQpD_Ih8q7I, .Y61Hh8votgHZzs6fsOmQQ, .weEQzf5OveNuvqSpUGQ2F, ._35eV4a3Obg89kE05EwXlOQ, .zDuKiGKd2bIY96obLG-xY, ._2O1M_31iF9yAEORnZjR5q- {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2; }

h1, ._6P1pAC-zOqNQpD_Ih8q7I {
  font-size: 2.1875rem; }

h2, .Y61Hh8votgHZzs6fsOmQQ {
  font-size: 1.75rem; }

h3, .weEQzf5OveNuvqSpUGQ2F {
  font-size: 1.53125rem; }

h4, ._35eV4a3Obg89kE05EwXlOQ {
  font-size: 1.3125rem; }

h5, .zDuKiGKd2bIY96obLG-xY {
  font-size: 1.09375rem; }

h6, ._2O1M_31iF9yAEORnZjR5q- {
  font-size: 0.875rem; }
```

You can see that HTML selectors are fine, but class selectors have been replaced with gibberish class names which is why the styles are not being applied to the HTML.

Let me know if you need more information. For the time being, I will revert back to 2.6.2.

**Comment by loopmode** [comment]
Looks like css modules to me. It's a feature of css-loader where classnames get hashed so they don't collide. There are two modes of this, an explicit mode where the user has to declare a css class as :local so it gets hashed, and an implicit mode where any classnames are treated as local unless you declare it :global.
If my theory is correct, what you see is that your custom classes have been all treated as local css modules.
@stevenkkim please try replacing e.g. `.my-class { ... }` by `:local(.my-class) { ... }` and check whether it fixes the problem. 
Once that is verified or falsified, we can see further.

**Comment by loopmode** [comment]
This is the relevant documentation: https://github.com/webpack-contrib/css-loader/blob/master/README.md#scope

I guess I introduced this new behaviour by setting `modules: true` in the config.
I guess we need to decide whether we leave that, and users have to opt out, or whether we remove flag and users must opt-in to modules.
I think the new releases dies finally support custom webpack config as a function, where users can mutate the config entirely - and thus can control even such detailed settings.

**Comment by stevenkkim** [comment]
Hi @loopmode, you are correct - adding `:global()` to classes that I have written fixes the problem. 

I am not using css modules, and as you can imagine, I do not want to re-write all my scss classes with `.global()`. Also I was unable to figure out from the css-loader documentation how to apply `:global()` to my bootstrap import:
```
$font-size-base: 0.875rem;
@import "../../node_modules/bootstrap/scss/bootstrap";
```

I don't know whether css modules is so popular now that is should be the default setting; it would seem from my perspective that the default should be `false` and let users opt in.

For the time being, can you tell me how to set css modules to `false` with a custom webpack config?

I tried adding
```
module.exports = {
  module: {
    rules: [
      {
        test: /\.css$/i,
        loader: 'css-loader',
        options: {
          modules: false,
        },
      },
    ],
  },
};
```
to both `webpack.renderer.additions.js` and to `webpack.main.additions.js` as well as adding to my `package.json`
```
  "electronWebpack": {
    "main": {
      "sourceDirectory": "src/electron/main",
      "webpackConfig": "webpack.main.additions.js"
    },
    "renderer": {
      "sourceDirectory": "src/electron/renderer",
      "webpackConfig": "webpack.renderer.additions.js"
    }
  }
```

But that did not work. I also tried changing `test: /\.css$/i,` to `test: /\.scss$/i,` but then I got errors with inline comments:
```
ERROR in ./src/styles/bootstrap.scss
  Module build failed (from ./node_modules/electron-webpack/node_modules/mini-css-extract-plugin/dist/loader.js):
  ModuleBuildError: Module build failed (from ./node_modules/css-loader/dist/cjs.js):
  CssSyntaxError

  (1:4) Unknown word

  > 1 | // https://getbootstrap.com/docs/4.3/getting-started/theming/
```

Thanks for your help!

**Comment by loopmode** [comment]
Hi @stevenkkim 

First of all, I just pushed [a fix](https://github.com/electron-userland/electron-webpack/commit/a34d8aeae737cafea3f749f4e0be04d8292779f6) that will be released with 2.7.2 (hopefully soon). 
It changes configuration so that the default is global CSS classes - and users must explicitly use `:local(.my-class) { ... }` in order to use CSS modules. I apologize for the inconvenience.

Next I'll fiddle a short-term solution for disabling CSS modules with 2.7.1 and post it here!

**Comment by loopmode** [comment]
Okay, so here's the fix for now.

in `package.json`:
```json
  "electronWebpack": {
    "renderer": {
      "webpackConfig": "webpack.renderer.js"
    }
  }
```

In `webpack.renderer.js`:
```js
module.exports = function(config) {
  const styleRules = config.module.rules.filter(rule =>
    rule.test.toString().match(/css|less|s\(\[ac\]\)ss/)
  )

  styleRules.forEach(rule => {
    const cssLoader = rule.use.find(use => use.loader === 'css-loader')
    cssLoader.options.modules = false
  })

  return config
}
```

Some notes:

- You only need this for the renderer - the main process doesn't handle stylesheets at all.

- Providing an object as webpack config won't do the trick - as it will be implicitly merged via `webpack-merge` [smart strategy](https://github.com/survivejs/webpack-merge#smart-merging). However, providing a function as webpack config is supported since 2.7.0 and this allows us to mutate the config  instead of only adding on-top

- The config function I posted should be self-explanatory, but to sum up what it does: it flips the switch on [this value](https://github.com/electron-userland/electron-webpack/blob/ee6a029573e445d80dfa5e6587daf5cb910ccf06/packages/electron-webpack/src/targets/RendererTarget.ts#L24), which will be [`"global"`](https://github.com/electron-userland/electron-webpack/blob/master/packages/electron-webpack/src/targets/RendererTarget.ts#L24) with the next release.

- For other users who might come here: If you already used a custom renderer config "the old way" - by exporting an object that implicitly got merged - you can still do the same thing: just do `const merge = require('webpack-merge')` and then inside the config function, in the end do `return merge.smart(config, myOldConfigObject)`. You can do manual changes to the config object and then still merge on top of it.

**Comment by loopmode** [comment]
Fix released with 2.7.2!
